src/courses/python/creating-a-basic-calculator-in-python-with-tkinter.md

```diff
@@ -0,0 +1,248 @@
+        # Chop off the ".0" at the end
+    updateValue()
+
+equalsButton = Button(master, text = '=', command = evaluate)
+```
+## The Complete Code
+That was a lot of code for just a little calculator. I've got over eighty lines of code here, which you can refer to whenever necessary.
+
+If you run your code and click on the buttons, you should see something pop up that looks like a calculator and calculates like a calculator. You've made a calculator! And you've also tapped into the power of this simple yet capable graphical user interface along the way.
+
+```python
+# A basic and functioning calculator using Tkinter in Python.
+from tkinter import *
+
+master = Tk()
+master.geometry('225x150')
+
+def Grid(Thing, Column, Row, Columnspan, Rowspan):
+    # A function I made that lets me write less code.
+    Thing.grid(column = Column, row = Row, columnspan = Columnspan, rowspan = Rowspan)
+
+displayedText = ''
+
+output = Label(master, text = '0')
+# Most calcs make 0 their default "nothing entered yet" screen.
+# Displays 0 at beginning of program.
+Grid(output, 0, 0, 5, 1)
+
+def updateValue():
+    output.config(text = str(displayedText))
+
+def append(value):
+    global displayedText
+    displayedText += value
+    updateValue()
+
+def clearText():
+    global displayedText
+    displayedText = '' # Get rid of whatever was in displayedText.
+    output.config(text = '0') # Reset text in output.
+
+def evaluate():
+    global displayedText
+    displayedText = str(float(eval(displayedText)))
+    if float(displayedText) == round(float(displayedText)):
+    # If the float's an integer
+        displayedText = str(round(float(displayedText))) 
+        # Chop off the ".0" at the end
+    updateValue()
+
+# HERE THERE BE BUTTONS.
+# In order, left to right, top to bottom.
+
+sevenButton = Button(master, text = '7', command = lambda : append('7'))
+Grid(sevenButton, 0, 1, 1, 1)
+
+eightButton = Button(master, text = '8', command = lambda : append('8'))
+Grid(eightButton, 1, 1, 1, 1)
+
+nineButton = Button(master, text = '9', command = lambda : append('9'))
+Grid(nineButton, 2, 1, 1, 1)
+
+plusButton = Button(master, text = '+', command = lambda : append('+'))
+Grid(plusButton, 3, 1, 1, 1)
+
+fourButton = Button(master, text = '4', command = lambda : append('4'))
+Grid(fourButton, 0, 2, 1, 1)
+
+fiveButton = Button(master, text = '5', command = lambda : append('5'))
+Grid(fiveButton, 1, 2, 1, 1)
+
+sixButton = Button(master, text = '6', command = lambda : append('6'))
+Grid(sixButton, 2, 2, 1, 1)
+
+minusButton = Button(master, text = '-', command = lambda : append('-'))
+Grid(minusButton, 3, 2, 1, 1)
+
+oneButton = Button(master, text = '1', command = lambda : append('1'))
+Grid(oneButton, 0, 3, 1, 1)
+
+twoButton = Button(master, text = '2', command = lambda : append('2'))
+Grid(twoButton, 1, 3, 1, 1)
+
+threeButton = Button(master, text = '3', command = lambda : append('3'))
+Grid(threeButton, 2, 3, 1, 1)
+

```

Comment:
Try using the **list in list** and **for loops** it will shorten the code pretty much for the button widgets.
Use something like this:-

numberPad = [['7','8','9','/'],
                        ['4','5','6',' *'],
                        ['1','2','3','-'],
                        ['.','0','DEL','+']]

for row in numberPad:
    columnNum = 0
    for column in row:
        numButton = ttk.Button(numeralPad, text = column, style = "_xyz_")
        numButton.bind("<Button-1>", _function_)
        numButton.grid(row = rowNum, column = columnNum, padx = x, pady = y,ipadx=ix, ipady=iy)
        columnNum +=1
    rowNum+=1

You can visit this code for reference [Calculator](https://github.com/SwaritBhardwaj/Scientific-Calculator)

Reply:
Thank you! This is an automated pull message sequence whenever people are working on tutorials. This tutorial is not live, and is not by me. I will forward this on to the author.